### PR TITLE
Add to doc for add_reference function.

### PR DESCRIPTION
**Pull request by elcubo**
Added description of :references option.

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb [hunk 1]
```diff
@@ -783,6 +783,8 @@ def index_name_exists?(table_name, index_name, default)
       #   Whether an additional +_type+ column should be added. Defaults to false.
       # [<tt>:null</tt>]
       #   Whether the column allows nulls. Defaults to true.
+      # [<tt>:references</tt>]
+      #   Name of the receiving table. Defaults to the table name of the target model.
```
Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb]: Not sure I understand this option? Can you share an example.

**Comment by elcubo** [comment]
@vipulnsward Added an example to explain the option.

File: activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb [hunk 2]
```diff
@@ -804,6 +806,10 @@ def index_name_exists?(table_name, index_name, default)
       #
       #   add_reference(:products, :supplier, foreign_key: {to_table: :firms})
       #
+      # ====== Create an administrator_id column as a reference to the users table
+      #
+      #   add_reference(:products, :administrator, references: :users)
```
Comment (vipulnsward) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb]: This is achieved by above `add_reference(:products, :supplier, foreign_key: {to_table: :firms})`, not with `references: :table_name`

**Comment by vipulnsward** [comment]
Thanks for the Pull Request @elcubo

**Pull request closed without merging**